src/utils/actions.js

```diff
@@ -82,19 +84,37 @@ const baseIgActions = {
       }
     }
 
-    const boolMap = action.condition.map(condition => {
-      if (condition.expression) {
-        const expression = condition.expression;
-        return evaluateExpression(expression, resources, variables);
-      } else {
-        // default true for non-applicability conditions
-        return true;
-      }
-    });
+    const boolMap = await Promise.all(
+      action.condition.map(async condition => {
+        if (condition.expression) {
+          const { expression, language } = condition.expression;
+
+          if (language === 'text/fhirpath') {
+            return isTrue(fhirpath.evaluate(resources, expression, variables));
+          } else if (language === 'text/cql') {
+            // Spec allows for multiple libraries, how can we tell which one is the main?
+            // Intentionally simplified and assuming the first library is the main
+            let library;
+            const planDefLib = context.planDefinition.library;
+            if (planDefLib?.length > 0) {
+              library = db.select(LIBRARYS, l => l.resource.id === planDefLib[0])[0].resource;
```

Comment:
Should we add another check here that `db.select(...)[0]` is defined? If there was an error adding the library to the database or a user deleted the library from the collection then `db.select(...)[0]` will return undefined and could cause an exception

Reply:
Makes sense, I added a check here and return false if the library is not in the database.

Reply:
I noticed here if `library` is undefined then the function returns false. Should that also be the behavior when `planDefLib.length <= 0`? Under that scenario `library` would still be undefined but the call to evaluate still happens.

Reply:
Good catch, I moved the `library` check after the `if` statement.